src/game/client/components/chat.cpp

```diff
@@ -800,12 +827,50 @@ void CChat::OnRender()
 			m_InputUpdate = false;
 		}
 
-		TextRender()->TextEx(&Cursor, m_Input.GetString()+m_ChatStringOffset, m_Input.GetCursorOffset()-m_ChatStringOffset);
-		static float MarkerOffset = TextRender()->TextWidth(0, 8.0f, "|", -1, -1.0f)/3;
-		CTextCursor Marker = Cursor;
-		Marker.m_X -= MarkerOffset;
-		TextRender()->TextEx(&Marker, "|", -1);
-		TextRender()->TextEx(&Cursor, m_Input.GetString()+m_Input.GetCursorOffset(), -1);
+		//render buffered text
+		if(m_Mode == CHAT_NONE) {
+			//truncate text to 29 chars
```

Comment:
Characters have different widths, so it would be best to cut the text based on width. 
- check if the rendered text is wider than a width limit.
- in case it is, render it with linewidth-width_of_"...", then render "..."

Reply:
the code does this, but is not very well written and one indizie is not right. I will update it

Reply:
You are right, I didn't understand it right